include/ada/url_aggregator-inl.h

```diff
@@ -206,6 +206,15 @@ inline void url_aggregator::update_base_pathname(const std::string_view input) {
   uint32_t difference = uint32_t(input.size()) - current_length;
   // The common case is current_length == 0.
   buffer.erase(components.pathname_start, current_length);
+  // next line is very uncommon and we should seek to optimize it accordingly.
+  if(has_empty_hostname() && !has_opaque_path && input.size()>1) {
+    // If url’s host is null, url does not have an opaque path, url’s path’s size is greater than 1,
+    // then append U+002F (/) followed by U+002E (.) to output.
+    buffer.insert(components.pathname_start, "/.");
+    components.pathname_start += 2;
+    if (components.search_start != url_components::omitted) { components.search_start += 2; }
+    if (components.hash_start != url_components::omitted) { components.hash_start += 2; }
```

Comment:
`difference += 2` would be sufficient instead of this changes, right?

Reply:
Correct. Let me make this change.